### PR TITLE
Refactoring as advised by @zero2one

### PR DESCRIPTION
**Pull request by Jelle-S**

### CODE REVIEW

**Comment by Jelle-S** [comment]
Not really a lot I can do about the codeclimate check fails I think

**Review by daften — APPROVED** [review]
Schoon :) Geen opmerkingen

File: src/Auth/Credentials.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace OpenBibIdApi\Auth;
+
+class Credentials implements CredentialsInterface
+{
+
```
Comment (zero2one) [inline src/Auth/Credentials.php]: No empty line between opening brace `{` and first method.

> **Is a remark for all classes.**

File: src/Auth/Environment.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    /**
+     * Get a production environment instance.
+     *
+     * @return \OpenBibIdApi\Auth\EnvironmentInterface
+     *   The production environment instance.
+     */
+    public static function production()
+    {
+        return new self('https://mijn.bibliotheek.be/openbibid/rest', 'prod');
```
Comment (zero2one) [inline src/Auth/Environment.php]: Use `static`:

```php
return new static(...);
```

File: src/Auth/Environment.php [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    /**
+     * Get a staging environment instance.
+     *
+     * @return \OpenBibIdApi\Auth\EnvironmentInterface
+     *   The staging environment instance.
+     */
+    public static function staging()
+    {
+        return new self('https://staging-mijn.bibliotheek.be/openbibid/rest', 'staging');
```
Comment (zero2one) [inline src/Auth/Environment.php]: Use `static`:

```php
return new static(...);
```

File: src/Auth/Environment.php [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRequestTokenUrl()
+    {
+        return $this->baseUrl. '/requestToken';
+    }
+
```
Comment (zero2one) [inline src/Auth/Environment.php:80]: 1 empty line between methods (instead of 2).

File: src/Auth/EnvironmentInterface.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    public function getBaseUrl();
+
+    /**
+     * Get the request token url.
+     *
+     * @return string
+     *   The request token url.
+     */
+    public function getRequestTokenUrl();
+
```
Comment (zero2one) [inline src/Auth/EnvironmentInterface.php:38]: 1 empty line between methods (instead of 2).

File: src/Consumer/BibConsumer.php [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+     */
+    protected $oauthConfig;
+
+    /**
+     *  The API Credentials.
+     *
+     * @var \OpenBibIdApi\Auth\CredentialsInterface
+     */
+    protected $credentials;
+
```
Comment (zero2one) [inline src/Consumer/BibConsumer.php:51]: 1 empty line between parameters and methods (instead of 2).

File: src/Service/LibraryService.php [hunk 7]
```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace OpenBibIdApi\Service;
+
+class LibraryService extends Service implements LibraryServiceInterface
```
Comment (zero2one) [inline src/Service/LibraryService.php]: Niet nodig om ook nog eens "implements LibraryServiceInterface" toe te voegen; Is reeds gebeurd door de Service class.

File: src/Service/ListService.php [hunk 8]
```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace OpenBibIdApi\Service;
+
+class ListService extends Service implements ListServiceInterface
```
Comment (zero2one) [inline src/Service/ListService.php:5]: Niet nodig om ook nog eens "implements LibraryServiceInterface" toe te voegen; Is reeds gebeurd door de Service class.

File: src/Service/ListServiceInterface.php [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+{
+
+    /**
+     * Get the lists of the current logged in user.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/ListServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

If another method, called in this method, throws an exception, no documentation about is required.

File: src/Service/ListServiceInterface.php [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+     * Get a list by id.
+     *
+     * @param string $listId
+     *   The list id.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/ListServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/ListServiceInterface.php [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+     * @param string $libId
+     *   The library id to search.
+     * @param string $itemId
+     *   The id of the item to search.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/ListServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/ListServiceInterface.php [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+     * Get all items of a list.
+     *
+     * @param string $listId
+     *   The list id.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/ListServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/OnlineCollectionServiceInterface.php [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+     * Get an online collection by its consumer key.
+     *
+     * @param string $consumerKey
+     *   The online collection consumer key.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/OnlineCollectionServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/OnlineCollectionServiceInterface.php [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+     * Get an online collection by id.
+     *
+     * @param string $collectionId
+     *   The online collection id.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/OnlineCollectionServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/OnlineCollectionServiceInterface.php [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+     * Get the current logged in user's permissions for an online collection.
+     *
+     * @param string $consumerKey
+     *   The online collection consumer key.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/OnlineCollectionServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

Comment (zero2one) [inline src/Service/OnlineCollectionServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/OnlineCollectionServiceInterface.php [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+     * @param string $consumerKey
+     *   The online collection consumer key.
+     * @param string $ipAddress
+     *   The IP address.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/OnlineCollectionServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/UserServiceInterface.php [hunk 17]
```diff
@@ -0,0 +1,141 @@
...
+{
+
+    /**
+     * Get the user info of the currently logged in user.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/UserServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/UserServiceInterface.php [hunk 18]
```diff
@@ -0,0 +1,141 @@
...
+    public function getUserInfo();
+
+    /**
+     * Get the user profile of the currently logged in user.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/UserServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/UserServiceInterface.php [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+    public function getUserProfile();
+
+    /**
+     * Get the available online collections for currently logged in user.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/UserServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/UserServiceInterface.php [hunk 20]
```diff
@@ -0,0 +1,141 @@
...
+    public function getUserAvailableOnlineCollections();
+
+    /**
+     * Get the library accounts of the currently logged in user.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/UserServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/UserServiceInterface.php [hunk 21]
```diff
@@ -0,0 +1,141 @@
...
+     * Get a library account by id for the currently logged in user.
+     *
+     * @param string $accountId
+     *   The id of a library account.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/UserServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/UserServiceInterface.php [hunk 22]
```diff
@@ -0,0 +1,141 @@
...
+     * Get the activities of a library account of the currently logged in user.
+     *
+     * @param string $accountId
+     *   The id of a library account.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/UserServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/UserServiceInterface.php [hunk 23]
```diff
@@ -0,0 +1,141 @@
...
+     * user.
+     *
+     * @param string $accountId
+     *   The id of the library account.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/UserServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/UserServiceInterface.php [hunk 24]
```diff
@@ -0,0 +1,141 @@
...
+    public function getUserLoanHistory($accountId);
+
+    /**
+     * Get the welcome messages of the currently logged in user.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/UserServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/UserServiceInterface.php [hunk 25]
```diff
@@ -0,0 +1,141 @@
...
+    public function getUserWelcomeMessages();
+
+    /**
+     * Get the libraries of the currently logged in user.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/UserServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

File: src/Service/UserServiceInterface.php [hunk 26]
```diff
@@ -0,0 +1,141 @@
...
+     * to a specific online collection.
+     *
+     * @param string $collectionKey
+     *   The online collection consumer key.
+     *
+     * @return \DOMDocument|null
+     *   A \DOMDocument containing the XML from the response, null if HTTP
+     *   status code 204 (No content) was returned.
+     *
+     * @throws BibApi\Exception\BibException
```
Comment (zero2one) [inline src/Service/UserServiceInterface.php]: @throws is only to be documented in the dochead if the method itself will throw that exception.

**Review by zero2one — CHANGES_REQUESTED** [review]

File: src/Service/LibraryService.php (hunk 7 above)
Comment (Jelle-S) [inline src/Service/LibraryService.php:5]: nee: Service implements ServiceInterface
LibraryService implements LibraryServiceInterface (= service om bibliotheken op te halen)

File: src/Service/ListService.php (hunk 8 above)
Comment (Jelle-S) [reply src/Service/ListService.php:5]: nee: Service implements ServiceInterface
ListService implements **List**ServiceInterface (= service om lijsten op te halen)

File: src/Storage/FileStorage.php [hunk 27]
```diff
@@ -2,49 +2,54 @@
...
+     * {@inheritdoc}
+     */
+    public function delete($key)
+    {
+        $key = $this->path . $key . '.txt';
+        if (file_exists($key)) {
+           unlink($key);
+        }
     }
-  }
```
Comment (zero2one) [inline src/Storage/FileStorage.php]: No extra empty line between last method and closing brace.

File: src/Storage/FileStorage.php [hunk 28]
```diff
@@ -2,49 +2,54 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete($key)
+    {
+        $key = $this->path . $key . '.txt';
+        if (file_exists($key)) {
+           unlink($key);
```
Comment (zero2one) [inline src/Storage/FileStorage.php]: One more space (12 spaces instead of 11).

File: src/Storage/SessionStorage.php [hunk 29]
```diff
@@ -2,47 +2,52 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete($key)
+    {
+        $key = $this->prefix . $key;
+        unset($_SESSION[$key]);
+    }
```
Comment (zero2one) [inline src/Storage/SessionStorage.php]: Remove empty line between last method and closing brace.

**Review by zero2one — CHANGES_REQUESTED** [review]

File: src/Auth/Environment.php (hunk 2 above)
Comment (Jelle-S) [inline src/Auth/Environment.php]: waarom static? Wanneer iemand overerft van Environment, willen we dan niet dat ze deze methode kunnen gebruiker zonder ze expliciet te moeten overschrijven?

File: src/Service/ListService.php (hunk 8 above)
Comment (zero2one) [reply src/Service/ListService.php:5]: ok

**Review by zero2one — APPROVED** [review]

**Pull request merged**